src/workers/MetricsWorker.ts

```diff
@@ -47,9 +56,12 @@ function workerThread() {
             intervalCounter.set(time - lastDumpTs);
             lastDumpTs = time;
             const dump = msg.substr('metricsdump:'.length);
-            res.statusCode = 200;
-            res.write(`${dump}\n${registry.metrics()}`);
-            res.end();
+            if (res.finished) {
+                // Sometimes a message will come in far too late because we've already
+                // sent an empty response. Drop it here.
```

Comment:
Drop it here? This isn't doing anything.

Reply:
Apparently I was half asleep and missed a return